src/main/scala/deduplication/Deduplication.scala

```diff
@@ -6,26 +6,63 @@ import java.util.concurrent.TimeoutException
 
 import scala.concurrent.duration._
 import scala.compat.java8.DurationConverters._
-import scala.collection.JavaConverters._
 
 import cats._
 import cats.implicits._
 import cats.effect._
-import cats.effect.implicits._
 
-import com.amazonaws.services.dynamodbv2.{AmazonDynamoDBAsync, AmazonDynamoDBAsyncClientBuilder}
-import com.amazonaws.services.dynamodbv2.model._
-import com.amazonaws.handlers._
-import org.scanamo._
-import org.scanamo.error.{DynamoReadError, NoPropertyOfType}
+import com.ovoenergy.comms.deduplication.model._
+import Deduplication._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.chrisdavenport.log4cats.StructuredLogger
 
-import model._
-import ScanamoHelpers._
+object Deduplication {
 
-/**
-  *
-  */
-trait Deduplication[F[_], ID] {
+  def nowF[F[_]: Functor: Clock] =
+    Clock[F]
+      .realTime(TimeUnit.MILLISECONDS)
+      .map(Instant.ofEpochMilli)
+
+  def processStatus[F[_]: Monad: Clock](
+      maxProcessingTime: FiniteDuration
+  )(p: Process[_, _]): F[ProcessStatus] = nowF[F].map { now =>
+
+    val isCompleted =
+      p.completedAt.isDefined
+
+    val isExpired = p.expiresOn
+      .map(_.instant)
+      .exists(_.isBefore(now))
+
+    val isTimeout = p.startedAt
+      .plus(maxProcessingTime.toJava)
+      .isBefore(now)
+
+    if (isCompleted && isExpired) {
```

Comment:
This booleans logic looks a bit weird if `isExpired` is true and everything else is false, it would result to `Running`(see the highlight line in the image)<img width="341" alt="Screenshot 2020-07-31 at 12 14 00" src="https://user-images.githubusercontent.com/36504063/89029964-66329e00-d327-11ea-874b-76a1e5f10c71.png">

Reply:
Interesting, this should never happen due to the fact that `expireOn` is only added when the process completed. This is a risk itself as if we have a bunch of processes that never get completed and never retried, they will pollute the table. By the other hand, if we assume that all the process needs to be done at least once, then it makes sense not expiring the non-completed one and I tend to agree on this second choice.

Reply:
But I agree that the logic is flawed, as at this point there is no guarantee that expires is not the only thing, I will change it

Reply:
Fixed